src/nunavut/lang/cpp/support/serialization.j2

```diff
@@ -930,6 +930,15 @@ inline {{typename_float_64}} const_bitspan::getF64()
     return tmp.fl;
 }
 
+template <typename T>
+void copyAssignInplace(T& dst, const T& src)
```

Comment:
This is incorrect. Please use `select_on_container_copy_construction` (https://en.cppreference.com/w/cpp/memory/polymorphic_allocator/select_on_container_copy_construction). See `variable_length_array` for example usage.

Reply:
@thirtytwobits Scott, is this what you are looking for?
https://github.com/OpenCyphal/nunavut/pull/349/commits/23b84f76d4fc0e75595efc02fde30140e3cef2fb
Honestly, I don't understand this part of c++ at all, and never encountered any book really describing such advanced topic of STL containers and their allocator traits and requirements.

Reply:
If you look at `variable_length_array` you'll see it does this:

```
: alloc_{std::allocator_traits<allocator_type>::select_on_container_copy_construction(rhs_alloc)}
```

Reply:
Additionally, in the copy assignment operator it adheres to the `propagate_on_container_copy_assignment` protocol defined by [allocator_traits](https://en.cppreference.com/w/cpp/memory/allocator_traits)

```
    // +----------------------------------------------------------------------+
    // | Allocator copy assignment
    // +----------------------------------------------------------------------+
    template <typename UAlloc>
    constexpr bool copy_assign_alloc(
        const UAlloc& rhs,
        typename std::enable_if_t<std::allocator_traits<UAlloc>::propagate_on_container_copy_assignment::value>* =
            nullptr) noexcept
    {
        static_assert(std::is_nothrow_copy_assignable<allocator_type>::value,
                      "The C++ standard requires anything adhering to propagate_on_container_copy_assignment to copy "
                      "without throwing.");
        alloc_ = rhs;
        return true;
    }

    template <typename UAlloc>
    constexpr bool copy_assign_alloc(
        const UAlloc& rhs,
        typename std::enable_if_t<!std::allocator_traits<UAlloc>::propagate_on_container_copy_assignment::value>* =
            nullptr) noexcept
    {
        (void) rhs;
        return false;
    }
```

Generally, the aliasing of memory here to do an in-place new on top of the current object seems dangerous. At the very least we'd need to understand if `std::launder` was needed in this case or not?

Reply:
Scott, I don't understand why in general I need to worry about all these `select_on_container...` and `propagate_on_container` stuff... All I need is along a variant (for union nunavut types) also store `allocator_`. This allocator is needed in case when value of the variant will be reassigned to some other variable size value (and so I need allocator to be able pass as the last parameter of constructor).

Reply:
>in-place new on top of the current object seems dangerous
I believe this is very close to what `optional::emplace` does

Reply:
ok